helix-term/src/commands.rs

```diff
@@ -4224,8 +4237,14 @@ pub mod insert {
             let current_line = text.char_to_line(pos);
             let line_start = text.line_to_char(current_line);
 
-            let continue_comment_token = continue_comment_tokens
-                .and_then(|tokens| comment::get_comment_token(text, tokens, current_line));
+            let continue_comment_token = comment::get_line_comment_token(
+                &cx.editor.syn_loader.load(),
+                syntax,
+                text,
+                doc_default_comment_token,
+                current_line,
+            )
+            .filter(|_| config.continue_comments);
```

Comment:
I think I mentioned this above but I can't find the comment. We should avoid doing the work in `comment::get_line_comment_token` when we know that `config.continue_comments` is false rather than filtering afterwards. I think we can also skip this work when `doc_default_comment_token` is `None`, so the pattern from the old code to treat both as `None` would make more sense